src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -487,6 +488,138 @@ def update_ag_redirect_configuration(cmd, instance, parent, item_name, redirect_
     return parent
 
 
+def create_ag_rewrite_rule_set(cmd, resource_group_name, application_gateway_name, item_name, no_wait=False):
+    ApplicationGatewayRewriteRuleSet = cmd.get_models(
+        'ApplicationGatewayRewriteRuleSet')
+    ncf = network_client_factory(cmd.cli_ctx).application_gateways
+    ag = ncf.get(resource_group_name, application_gateway_name)
+    new_set = ApplicationGatewayRewriteRuleSet(name=item_name)
+    _upsert(ag, 'rewrite_rule_sets', new_set, 'name')
+    if no_wait:
+        return sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag)
+    parent = sdk_no_wait(no_wait, ncf.create_or_update, resource_group_name, application_gateway_name, ag).result()
+    return _find_child_item(parent, item_name,
+                            path='rewrite_rule_sets', key_path='name')
+
+
+def update_ag_rewrite_rule_set(instance, parent, item_name):
```

Comment:
What does this do?

Reply:
Is this to enforce that item_name is required?

Reply:
No. It is because the rule set command is defined in a loop with a bunch of other app gateway subresources and that means it expects an update. There are no convenience arguments for rewrite rule sets, which is why it doesn't do anything.  Essentially, it's a no-op.